app/dao/RouteDao.scala

```diff
@@ -126,6 +129,17 @@ class RouteDao @Inject()(routeRepo: RouteRepo)(implicit ec: ExecutionContext)
     routeRepo.getRouteIdFor(userId, DateTime.now())
   }
 
+  def getRoutes(id: Long): Future[util.List[common.Route]] = {
+    routeRepo.getRoutesFor(id)
+      .map {
+        seq =>
+          seq.groupBy(_._1.id).values.map {
+            rows =>
+              composeRoute(rows).get
+          }.toList.sortBy(_.getRouteDate).reverse.asJava
+      }
```

Comment:
I'd play with some newlines here for readability
(and add comment that we certainly have route under `Option` since we grouped by its id)